File: MCServerWebWrapper.Server/Data/Models/MinecraftServer.cs
using MCServerWebWrapper.Server.Models;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace MCServerWebWrapper.Server.Data.Models
{
	public class MinecraftServer
	{
		[BsonId]
		[BsonRepresentation(BsonType.ObjectId)]
		public string Id { get; set; }
        public string Name { get; set; }
        public bool IsRunning { get; set; }
        public int TimesRan { get; set; }
        public int PlayersCurrentlyConnected { get; set; }
        public IList<PlayerCountChange> PlayerCountChanges { get; set; }
        public int ContainerId { get; set; }
        public int MaxRamMB { get; set; }
        public int InitRamMB { get; set; }
        public string ServerPath { get; set; }
        public IList<string> Operators { get; set; }
        public IList<string> WhitelistedPlayers { get; set; }
        public string Type { get; set; }
        public IDictionary<string, string> TypeSpecificEnvVariables { get; set; }
        public string Version { get; set; }
        public Uri ServerIconUri { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime DateLastStarted { get; set; }
        public DateTime DateLastStopped { get; set; }
        public TimeSpan TotalUpTime { get; set; }
        public Properties Properties { get; set; }
        public IList<Output> Logs { get; set; }
	}
}

File: MCServerWebWrapper.Server/ClientApp/src/app/models/minecraft-server.ts
export interface MinecraftServer {
  id: string;
  name: string;
  maxRamMB: number;
  minRamMB: number;
  isRunning: boolean;
  timesRan: number;
  playersCurrentlyConnected: number
  playerCountChanges: PlayerCountChange[];
  dateCreated: Date;
  dateLastStarted: Date;
  dateLastStopped: Date;
  totalUpTimeMs: number;
  percentUpTime: number;
  latestLogs: Output[];
}

interface Output {
  timeStamp: Date;
  line: string;
}

interface PlayerCountChange {
  timestamp: Date;
  playersCurrentlyConnected: number;
  triggeredByUsername: string;
  isJoin: boolean;
}
